flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -1061,6 +1063,20 @@ void markFinished(Map<String, Accumulator<?, ?>> userAccumulators, IOMetrics met
 							IntermediateResultPartition[] allPartitions = finishedPartition
 									.getIntermediateResult().getPartitions();
 
+							PartitionTable<ResourceID> partitionTable = vertex.getExecutionGraph().getPartitionTable();
```

Comment:
Maybe better to put the following logics in a separate method to make markFinished shorter.